File: CameraNotifier/Services/SlackNotifier/SlackNotifier.cs
using System;
using System.Collections.Specialized;
using System.IO;
using System.Net;
using System.Threading.Tasks;
using Microsoft.Extensions.Options;
using Serilog;

namespace CameraNotifier.Services.SlackNotifier
{
    internal class SlackNotifier : ISlackNotifier
    {
        private const string SlackUploadUri = "https://slack.com/api/files.upload";

        private readonly SlackNotifierOptions _options;

        public SlackNotifier(IOptions<SlackNotifierOptions> options)
        {
            _options = options.Value;
            Log.Logger.Information("Initializing SlackNotifier");
        }

        public void SendNotification(string text, string imageFilePath)
        {
            UploadFilesToChannel(_options.Channel, Path.GetFileName(imageFilePath),
                imageFilePath, text).Wait();
        }

        public async Task UploadFilesToChannel(string channel, string fileName, string filePath, string comment)
        {
            var nameValueCollection = new NameValueCollection()
            {
                { "token", _options.APIKey },
                { "channels", channel },
                { "filename", fileName },
                { "initial_comment", comment}
            };

            var client = new WebClient() {QueryString = nameValueCollection};
            await client.UploadFileTaskAsync(new Uri(SlackUploadUri), filePath);
        }
    }
}

File: python/slack_notifier.py
from slack_sdk import WebClient
from slack_sdk.errors import SlackApiError
import os


class SlackNotifier:
    def __init__(self, config):
        self.api_key = config['SlackNotifier']['api_key']
        self.channel = config['SlackNotifier']['channel']
        self.client = WebClient(token=self.api_key)
        print("Initializing SlackNotifier")

    def send_notification(self, text, image_file_path):
        """Send a text notification with an image attachment to Slack"""
        try:
            # Upload file
            response = self.client.files_upload(
                channels=self.channel,
                file=image_file_path,
                initial_comment=text
            )
            
            return response
        except SlackApiError as e:
            print(f"Failed to send Slack notification: {e.response['error']}")